### PR TITLE
Add user admin flag and deletion reasons to the DB

### PR DESCRIPTION
**Pull request by rosalieper**
https://phabricator.wikimedia.org/T347865
The admins here are WMDE wikibase cloud team

### CODE REVIEW

File: database/migrations/2024_05_15_042959_add_admin_role_to_users.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->boolean('is_admin')->default(false);
```
Comment (m90) [inline database/migrations/2024_05_15_042959_add_admin_role_to_users.php:15]: Idea: if we made this an `int`, we could enumerate "levels" of admins, i.e. `0` would be a plain user, `1` would be a "super admin" and when needed, we could start adding more values to model some sort of permissions.

Comment (m90) [reply database/migrations/2024_05_15_042959_add_admin_role_to_users.php:15]: I have no objections against using a boolean for now though, so feel free to disregard this.

File: app/User.php [hunk 2]
```diff
@@ -74,6 +74,7 @@ class User extends Authenticatable implements MustVerifyEmail
         'trial_ends_at',
         'card_brand',
         'card_last_four',
+        'is_admin',
```
Comment (tarrow) [inline app/User.php:77]: I think this should probably be in "fillable" rather than hidden. What do you think?

File: app/Wiki.php [hunk 3]
```diff
@@ -52,6 +52,7 @@ class Wiki extends Model
         'domain',
         'description',
         'is_featured',
+        'wiki_deletion_reason',
```
Comment (tarrow) [inline app/Wiki.php:55]: looks good to me

**Review by tarrow — APPROVED** [review]
nice, let's see this merged then :)

**Pull request merged**